nestkernel/conn_builder.cpp

```diff
@@ -1101,14 +1101,28 @@ nest::FixedInDegreeBuilder::connect_()
       {
         // check whether the target is on this mpi machine
         if ( not kernel().node_manager.is_local_gid( *tgid ) )
+        {
+          for ( long j = 0; j < indegree_; ++j )
+          {
+            // skip array parameters handled in other virtual processes
+            skip_conn_parameter_( tid );
```

Comment:
Looping over `skip_conn_parameter_()` can become very inefficient due to nested loops and function calls, including to `ConnParameter::skip()`, which requires a vtable lookup. I think it would be far more efficient to pass the number of elements to skip to `skip_conn_parameter()` as second argument (default value 1), pass that through to `ConnParamter::skip()` and then use the most efficient implementation "down there". For `Array*Parameter::skip()`, that probably would just mean replacing `*next_[ tid ]++` with `next_[ tid ] += j` (the derefercing in the current code makes no sense).